acapy_agent/did/cheqd/cheqd_manager.py

```diff
@@ -107,3 +116,142 @@ async def register(self, options: dict) -> dict:
             "did": did,
             "verkey": verkey,
         }
+
+    async def update(
+        self, did: str, didOperation: str, didDoc: dict, options: dict
+    ) -> dict:
+        """Update a Cheqd DID."""
+
+        async with self.profile.session() as session:
+            try:
+                wallet = session.inject(BaseWallet)
+                if not wallet:
+                    raise web.HTTPForbidden(reason="No wallet available")
+
+                # Check if did is valid
+                await self.resolver.resolve(self.profile, did)
+                # request deactivate did
+                update_request_res = await self.registrar.update(
+                    {
+                        "did": did,
+                        "didDocumentOperation": [didOperation],
+                        "didDocument": [didDoc],
+                    }
+                )
+
+                job_id: str = update_request_res.get("jobId")
+                did_state = update_request_res.get("didState")
+
+                if did_state.get("state") == "action":
+                    sign_req: dict = did_state.get("signingRequest")[0]
```

Comment:
we need to update the signing logic to sign with all the kid's requested instead of assuming only one key for a did. 

We can create a common function for create, update and deactivate which loops through all n signing request and return n signatures.

Reply:
Done, new function added